### PR TITLE
Add basic authentication [ LABS-223 ]

### PR DESCRIPTION
**Pull request by fvosburgh**
I think this is pretty straightforward. Basic authentication requires an authentication header that is base64 encoded. This should be upgraded to a better security posture in the future, but it should suffice for now.

### CODE REVIEW

File: spec/controller/scan_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+
+describe ScansController do
+  describe 'POST', type: :request do
```
Comment (fvosburgh) [inline spec/controller/scan_controller_spec.rb]: I wasn't sure if I should add another `describe` block noting that these are testing for authentication only. I figured that the wording in the context and example blocks would make it clear.

Comment (wcmonty) [inline spec/controller/scan_controller_spec.rb]: I think this is fine for now and we can add a context later if that clarifies things.

File: spec/controller/scan_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    let(:payload) {{ 'result': 'passed', 'event':  'deploy', 'commit': { } }}
+
+    context 'without credentials' do
+      it 'does not authenticate' do
+        post '/', scan: payload
+        expect(last_response.status).to eql(401)
+      end
+    end
+
+    context 'with credentails' do
```
Comment (wcmonty) [inline spec/controller/scan_controller_spec.rb]: s/credentails/credentials/

File: app/concerns/commit_formatter.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 module CommitFormatter
   def format_commit(commit)
-    commit['id'] = format_commit_id commit['id']
-    commit
+    if commit
```
Comment (wcmonty) [inline app/concerns/commit_formatter.rb]: What about the following:

```
def format_commit(commit)
  commit.tap { |c| c['id'] =  format_commit_id(c['id']) if c } 
end
```

Also, this code is pointing to the fact that a commit should be a objectified.  We can defer that, but I consider the repetition of the word `commit` to be a code smell.

File: README.md [hunk 4]
```diff
@@ -13,6 +13,12 @@ the total number of defects it has by leveraging Rigor Optimization scans.
 * Deploy the app to Heroku with the deploy button above or the host of your choice.
 * Set the environment variables:
 ```
+# Username you want for authentication
+PERFUSER
```
Comment (wcmonty) [inline README.md]: We might want to add an initializer that will throw an error if the required ENV vars are not in place.  Might make troubleshooting easier.  Again, we can defer.

**Comment by wcmonty** [comment]
@fvosburgh Looks good.  Some minor tweaks/suggestions & back to you.

File: spec/controller/scan_controller_spec.rb (hunk 2 above)
Comment (fvosburgh) [inline spec/controller/scan_controller_spec.rb]: Addressed in f968ceb2a4f518cf9c27e7302c4da36dcb700867

File: app/concerns/commit_formatter.rb (hunk 3 above)
Comment (fvosburgh) [inline app/concerns/commit_formatter.rb]: Addressed in e7ad502813be71e60cce9db070b8f1fd026ef91e

**Pull request merged**